crates/database/src/settlement_observations.rs

```diff
@@ -0,0 +1,84 @@
+use {crate::events::EventIndex, bigdecimal::BigDecimal, sqlx::PgConnection};
+
+#[derive(Debug, PartialEq, sqlx::FromRow)]
+pub struct Observation {
+    pub gas_used: BigDecimal,
+    pub effective_gas_price: BigDecimal,
+    pub surplus: BigDecimal,
+    pub fee: BigDecimal,
+    pub block_number: i64,
+    pub log_index: i64,
+}
+
+pub async fn upsert(
```

Comment:
Is `upsert` the desired term here? I guess it is intended to mean both "insert" and "update"? Makes sense then.

Reply:
Yes I saw that we already used `upsert` for similar logic so just followed for consistency.